sources/packages/backend/apps/api/src/route-controllers/cas-supplier/_tests_/cas-supplier.aest.controller.retryCASSupplier.e2e-spec.ts

```diff
@@ -0,0 +1,126 @@
+  it("Should throw Unprocessable Entity Error when retry CAS supplier is requested for a student who already has a CAS supplier in Pending supplier verification status.", async () => {
+    // Arrange
+    // Create a CAS Supplier for the student with Pending supplier verification status.
+    const casSupplier = await saveFakeCASSupplier(db, undefined, {
+      initialValues: {
+        supplierStatus: SupplierStatus.PendingSupplierVerification,
+      },
+    });
+    const student = await saveFakeStudent(db.dataSource, undefined, {
+      initialValue: {
+        casSupplier: casSupplier,
+      },
+    });
+    const endpoint = `/aest/cas-supplier/student/${student.id}/retry`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        statusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+        message:
+          "There is already a CAS Supplier for this student in Pending supplier verification status.",
+        error: "Unprocessable Entity",
+      });
+  });
+
+  it("Should create CAS supplier when retry CAS supplier is requested for a student who does not have a latest CAS supplier in Pending supplier verification status.", async () => {
+    // Arrange
+    // Create a CAS Supplier for the student with Manual Intervention status.
+    const fakeCASSupplier = await saveFakeCASSupplier(db, undefined, {
+      initialValues: {
+        supplierStatus: SupplierStatus.ManualIntervention,
+      },
+    });
+    const student = await saveFakeStudent(db.dataSource, undefined, {
+      initialValue: {
+        casSupplier: fakeCASSupplier,
+      },
+    });
+    const endpoint = `/aest/cas-supplier/student/${student.id}/retry`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    let responseCasSupplierId: number;
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        responseCasSupplierId = +response.body.id;
+        expect(responseCasSupplierId).toEqual(expect.any(Number));
+      });
+
+    const casSupplier = await db.casSupplier.findOne({
+      where: {
+        id: responseCasSupplierId,
+      },
+      order: {
+        createdAt: "DESC",
+      },
+    });
+
+    expect(casSupplier).toEqual({
+      id: responseCasSupplierId,
+      supplierStatus: SupplierStatus.PendingSupplierVerification,
+      isValid: false,
+      createdAt: expect.any(Date),
+      lastUpdated: null,
+      supplierStatusUpdatedOn: expect.any(Date),
+      errors: null,
+      status: null,
+      studentProfileSnapshot: null,
+      supplierAddress: null,
+      supplierName: null,
+      supplierNumber: null,
+      supplierProtected: null,
+      updatedAt: expect.any(Date),
+    });
+  });
```

Comment:
For the future, there was no ask to check every single `null` value created, but it can be kept if the below comment is adjusted.
https://github.com/bcgov/SIMS/pull/4053/files#r1879007127